lib/async/enumerator.rb

```diff
@@ -17,10 +18,22 @@ class Enumerator
 
     # Creates async wrapper for enumerable.
     # @param enumerable [Enumerable] Object to wrap
-    # @param max_fibers [Integer, nil] Concurrency limit
-    def initialize(enumerable, max_fibers: nil)
+    # @param config [Config, nil] Configuration object
+    # @param kwargs [Hash] Configuration options (max_fibers, etc.)
+    def initialize(enumerable = [], config = nil, **kwargs)
       @enumerable = enumerable
-      @max_fibers = max_fibers
+
+      # Handle the common case of Enumerator.new(enumerable, max_fibers: n)
+      if config.is_a?(Hash) && !config.is_a?(Config)
+        kwargs = config
+        config = nil
+      end
```

Comment:
if config is a Hash, it can't also be a config -- what are we trying to do here?